### PR TITLE
Implement full copy non evm proof

### PR DESCRIPTION
**Pull request by prin-r**
## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

**Review by evilpeach — APPROVED** [review]

File: scan/src/utils/proof/NonEVMProof.re [hunk 1]
```diff
@@ -16,134 +15,362 @@ type response_packet_t = {
...
+  | BlockHeaderMerkleParts(block_header_merkle_parts_t)
+  | Signature(tm_signature_t)
+  | Signatures(list(tm_signature_t))
+  | Proof(proof_t);
+
+let rec encode =
+  fun
+  | RequestPacket({clientID, oracleScriptID, calldata, askCount, minCount}) => {
+      Obi.encode(
+        {j|{clientID: string, oracleScriptID: u64, calldata: bytes, askCount: u64, minCount: u64}/{_:u64}|j},
```
Comment (tansawit) [inline scan/src/utils/proof/NonEVMProof.re]: Use normal string here?

File: scan/src/utils/proof/NonEVMProof.re [hunk 2]
```diff
@@ -16,134 +15,362 @@ type response_packet_t = {
...
-        resolveTimne: resolveTime |> Belt_Option.getExn |> MomentRe.Moment.toUnix,
-        resolveStatus: resolveStatus |> resolveStatusToInt,
-        result: result |> Belt_Option.getExn,
-      },
-    )
-  };
-};
+    }) => {
+      Obi.encode(
+        {j|{clientID: string, requestID: u64, ansCount: u64, requestTime: u64, resolveTime: u64, resolveStatus: u32, result: bytes}/{_:u64}|j},
```
Comment (tansawit) [inline scan/src/utils/proof/NonEVMProof.re]: ditto?

File: scan/src/utils/proof/NonEVMProof.re [hunk 3]
```diff
@@ -16,134 +15,362 @@ type response_packet_t = {
...
+          {fieldName: "requestTime", fieldValue: requestTime |> string_of_int},
+          {fieldName: "resolveTime", fieldValue: resolveTime |> string_of_int},
+          {fieldName: "resolveStatus", fieldValue: resolveStatus |> string_of_int},
+          {fieldName: "result", fieldValue: result |> JsBuffer.toHex(~with0x=true)},
+        |],
+      );
+    }
+  | IAVLMerklePath({isDataOnRight, subtreeHeight, subtreeSize, subtreeVersion, siblingHash}) => {
+      Obi.encode(
+        {j|{isDataOnRight: u8, subtreeHeight: u8, subtreeSize: u64, subtreeVersion: u64, siblingHash: bytes}/{_:u64}|j},
```
Comment (tansawit) [inline scan/src/utils/proof/NonEVMProof.re]: ditto

File: scan/src/utils/proof/NonEVMProof.re [hunk 4]
```diff
@@ -16,134 +15,362 @@ type response_packet_t = {
...
+          lastBlockIDAndOther,
+          nextValidatorHashAndConsensusHash,
+          lastResultsHash,
+          evidenceAndProposerHash,
+        |]),
+      );
+    }
+  | Signature({r, s, v, signedPrefixSuffix, signedDataSuffix}) => {
+      Obi.encode(
+        {j|{r: bytes, s: bytes, v: u8, signedPrefixSuffix: bytes, signedDataSuffix: bytes}/{_:u64}|j},
```
Comment (tansawit) [inline scan/src/utils/proof/NonEVMProof.re]: ditto

File: scan/src/utils/proof/NonEVMProof.re [hunk 5]
```diff
@@ -16,134 +15,362 @@ type response_packet_t = {
...
+            fieldValue: signedPrefixSuffix |> JsBuffer.toHex(~with0x=true),
+          },
+          {
+            fieldName: "signedDataSuffix",
+            fieldValue: signedDataSuffix |> JsBuffer.toHex(~with0x=true),
+          },
+        |],
+      );
+    }
+  | Signatures(list_of_tm_signatures) => {
```
Comment (tansawit) [inline scan/src/utils/proof/NonEVMProof.re]: rename/shorten to `tm_signatures`?

File: scan/src/utils/proof/NonEVMProof.re [hunk 6]
```diff
@@ -16,134 +15,362 @@ type response_packet_t = {
...
+        [|
+          {fieldName: "isDataOnRight", fieldValue: isDataOnRight ? "1" : "0"},
+          {fieldName: "subtreeHeight", fieldValue: subtreeHeight |> string_of_int},
+          {fieldName: "subtreeSize", fieldValue: subtreeSize |> string_of_int},
+          {fieldName: "subtreeVersion", fieldValue: subtreeVersion |> string_of_int},
+          {fieldName: "siblingHash", fieldValue: siblingHash |> JsBuffer.toHex(~with0x=true)},
+        |],
+      );
+    }
+  | IAVLMerklePaths(list_of_iavl_merkle_paths) => {
```
Comment (tansawit) [inline scan/src/utils/proof/NonEVMProof.re]: Rename/shorten to `iavl_merkle_paths`?

**Review by tansawit — CHANGES_REQUESTED** [review]

File: scan/src/utils/proof/NonEVMProof.re (hunk 1 above)
Comment (prin-r) [reply scan/src/utils/proof/NonEVMProof.re]: done

File: scan/src/utils/proof/NonEVMProof.re (hunk 2 above)
Comment (prin-r) [reply scan/src/utils/proof/NonEVMProof.re]: done

File: scan/src/utils/proof/NonEVMProof.re (hunk 3 above)
Comment (prin-r) [reply scan/src/utils/proof/NonEVMProof.re]: done

File: scan/src/utils/proof/NonEVMProof.re (hunk 4 above)
Comment (prin-r) [reply scan/src/utils/proof/NonEVMProof.re]: done

File: scan/src/utils/proof/NonEVMProof.re (hunk 5 above)
Comment (prin-r) [reply scan/src/utils/proof/NonEVMProof.re]: done

File: scan/src/utils/proof/NonEVMProof.re (hunk 6 above)
Comment (prin-r) [reply scan/src/utils/proof/NonEVMProof.re]: done

**Comment by evilpeach** [comment]
Can we merge after revamped GuanYu version done?

**Comment by evilpeach** [comment]
@pzshine Can you check about design?

**Review by tansawit — APPROVED** [review]

**Pull request merged**